test/Calendar.test.js

```diff
@@ -6,34 +6,172 @@ const fixtures = require("./fixtures/index");
 
 const CALENDAR_DOMAIN = "http://example.com";
 const CALENDAR_PATH = "/cal.php/calendars/user/calendar_name/";
+const config = {
+    auth: {
+        user: "username",
+        pass: "password"
+    },
+    uri: CALENDAR_DOMAIN + CALENDAR_PATH
+};
 
 describe("Calendar", () => {
 
+    it("should throw if config object is not passed", () => {
+        const Calendar = createCalendar(() => {
```

Comment:
From this code I do not understand what exactly you're passing to `createCalendar`.

Please name the dependency:

```javascript
const request = Promise.resolve({
  // would be nice to see how the data looks like if it does not exceed screen size
});
const Calendar = createCalendar(request);
```